src/main/java/com/gregtechceu/gtceu/api/recipe/GTRecipe.java

```diff
@@ -54,6 +57,16 @@ public class GTRecipe implements net.minecraft.world.item.crafting.Recipe<Contai
     public int subtickParallels = 1;
     public int batchParallels = 1;
     public int ocLevel = 0;
+    /**
+     * Populated after the inputs are already consumed, but the recipe didn't start yet
+     * For use in {@link GTRecipe#itemOutputModifier}
+     */
+    public List<ItemStack> itemInputs = new ArrayList<>();
```

Comment:
this should be generic to all input types.
```suggestion
    public Map<RecipeCapability<?>, List<Object>> realInputs = new Reference2ObjectOpenHashMap<>();
```